ConsentViewController/Classes/SourcePointClient/SourcepointClientCoordinator.swift

```diff
@@ -874,89 +870,89 @@ class SourcepointClientCoordinator: SPClientCoordinator {
 
     func postChoice(
         _ action: SPAction,
-        postPayloadFromGetCall: ChoiceAllResponse.GDPR.PostPayload?,
-        handler: @escaping (Result<GDPRChoiceResponse, SPError>) -> Void
+        postPayloadFromGetCall: SPMobileCore.ChoiceAllResponse.GDPRPostPayload?,
+        handler: @escaping (Result<SPMobileCore.GDPRChoiceResponse, SPError>) -> Void
     ) {
         spClient.postGDPRAction(
             actionType: action.type,
-            body: GDPRChoiceBody(
+            request: GDPRChoiceRequest(
                 authId: authId,
                 uuid: state.gdpr?.uuid,
                 messageId: action.messageId,
                 consentAllRef: postPayloadFromGetCall?.consentAllRef,
                 vendorListId: postPayloadFromGetCall?.vendorListId,
-                pubData: action.encodablePubData,
-                pmSaveAndExitVariables: action.pmPayload,
+                pubData: JsonKt.encodeToJsonObject(action.encodablePubData.toCore()),
```

Comment:
That's such a clever solution. I think I looked for a way to pass a json object to the core module and I couldn't find it!